integration/spark2/src/main/scala/org/apache/spark/sql/parser/CarbonSparkSqlParser.scala

```diff
@@ -169,220 +128,45 @@ class CarbonHelperSqlAstBuilder(conf: SQLConf,
-        table.getFactTable.getTableProperties.put("_external", "false")
-      } else {
-        table.getFactTable.getTableProperties.put("_external", "true")
-        table.getFactTable.getTableProperties.put("_filelevelformat", "false")
-      }
-      // setting local dictionary for all string coloumn for external table
-      var isLocalDic_enabled = table.getFactTable.getTableProperties
-        .get(CarbonCommonConstants.LOCAL_DICTIONARY_ENABLE)
-      if (null == isLocalDic_enabled) {
-        table.getFactTable.getTableProperties
-          .put(CarbonCommonConstants.LOCAL_DICTIONARY_ENABLE,
-            CarbonCommonConstants.LOCAL_DICTIONARY_ENABLE_DEFAULT)
-      }
-      isLocalDic_enabled = table.getFactTable.getTableProperties
-        .get(CarbonCommonConstants.LOCAL_DICTIONARY_ENABLE)
-      if (CarbonScalaUtil.validateLocalDictionaryEnable(isLocalDic_enabled) &&
-          isLocalDic_enabled.toBoolean) {
-        val allcolumns = table.getFactTable.getListOfColumns
-        for (i <- 0 until allcolumns.size()) {
-          val cols = allcolumns.get(i)
-          if (cols.getDataType == DataTypes.STRING || cols.getDataType == DataTypes.VARCHAR) {
-            cols.setLocalDictColumn(true)
-          }
-          allcolumns.set(i, cols)
-        }
-        table.getFactTable.setListOfColumns(allcolumns)
-      }
-
-      table
-    } else {
-      // prepare table model of the collected tokens
-      val tableModel: TableModel = parser.prepareTableModel(
-        ifNotExists,
-        convertDbNameToLowerCase(tableIdentifier.database),
-        tableIdentifier.table.toLowerCase,
-        fields,
-        partitionFields,
-        tableProperties,
-        bucketFields,
-        isAlterFlow = false,
-        false,
-        tableComment)
-      TableNewProcessor(tableModel)
-    }
-    tableInfo.setTransactionalTable(isTransactionalTable)
-    selectQuery match {
-      case query@Some(q) =>
-        CarbonCreateTableAsSelectCommand(
-          tableInfo = tableInfo,
-          query = query.get,
-          ifNotExistsSet = ifNotExists,
-          tableLocation = tablePath)
-      case _ =>
-        CarbonCreateTableCommand(
-          tableInfo = tableInfo,
-          ifNotExistsSet = ifNotExists,
-          tableLocation = tablePath,
-          external)
-    }
-  }
-
-  private def validateStreamingProperty(carbonOption: CarbonOption): Unit = {
-    try {
-      carbonOption.isStreaming
-    } catch {
-      case _: IllegalArgumentException =>
-        throw new MalformedCarbonCommandException(
-          "Table property 'streaming' should be either 'true' or 'false'")
-    }
+    val extraTableTuple = (cols, external, tableIdentifier, ifNotExists, colNames, tablePath,
+      tableProperties, properties, partitionByStructFields, partitionFields,
+      parser, sparkSession, selectQuery)
+    CarbonSparkSqlParserUtil
+      .createCarbonTable(createTableTuple, extraTableTuple)
   }
 
   private def validatePartitionFields(
```

Comment:
Now this method has only 1 line of code. So, we can move this code line to caller and avoid this method completely.